File: src/Oisys.Web/Models/Item.cs
using System.Collections.Generic;

namespace OisysNew.Models
{
    /// <summary>
    /// <see cref="Item"/> class Item/Inventory object.
    /// </summary>
    public class Item : SoftDeletableModel
    {
        /// <summary>
        /// Gets or sets property Code.
        /// </summary>
        public string Code { get; set; }

        /// <summary>
        /// Gets or sets property Name.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets property Description.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Gets or sets a snapshot of the item's quantity.
        /// </summary>
        public int Quantity { get; set; }

        /// <summary>
        /// Gets or sets property Unit.
        /// </summary>
        public string Unit { get; set; }

        /// <summary>
        /// Gets or sets property MainPrice.
        /// </summary>
        public decimal MainPrice { get; set; }

        /// <summary>
        /// Gets or sets property NEPrice.
        /// Also known as Nueva Ecija Price.
        /// </summary>
        public decimal NEPrice { get; set; }

        /// <summary>
        /// Gets or sets property WalkInPrice.
        /// </summary>
        public decimal WalkInPrice { get; set; }

        /// <summary>
        /// Gets or sets property Weight.
        /// </summary>
        public string Weight { get; set; }

        /// <summary>
        /// Gets or sets property Thickness.
        /// </summary>
        public string Thickness { get; set; }

        /// <summary>
        /// Gets or sets property CategoryId.
        /// </summary>
        public long CategoryId { get; set; }

        /// <summary>
        /// Gets or sets property Category.
        /// </summary>
        public virtual Category Category { get; set; }

        /// <summary>
        /// Gets or sets transaction history for the item.
        /// </summary>
        public virtual ICollection<ItemTransactionHistory> TransactionHistory { get; set; }
    }
}

File: src/Oisys.Web/ClientApp/src/app/shared/models/item.ts
import { Category } from "./category";

export class Item {
  public id: number;
  public code: string;
  public name: string;
  public description: string;
  public categoryId: number;
  public categoryName: string;
  public quantity: number;
  public stockQuantity: number;
  public unit: string;
  public weight: string;
  public thickness: string;
  public mainPrice: number;
  public walkInPrice: number;
  public nePrice: number;
  public rowVersion: string;

  private _category: Category;
  get category() {
    return this._category;
  }
  set category(category: Category) {
    if (category) {
      this._category = category;
      this.categoryId = category.id;
    }
  }

  constructor();
  constructor(item: Item);
  constructor(item?: any) {
    this.id = item && item.id || 0;
    this.code = item && item.code || '';
    this.name = item && item.name || '';
    this.description = item && item.description || '';
    this.rowVersion = item && item.rowVersion || '';

    this.categoryId = item && item.categoryId || 0;
    this.categoryName = item && item.categoryName || '';

    this.quantity = item && item.quantity || 0;
    this.stockQuantity = item && item.stockQuantity || 0;
    this.unit = item && item.unit || '';
    this.weight = item && item.weight || '';
    this.thickness = item && item.thickness || '';

    this.mainPrice = item && item.mainPrice || 0;
    this.walkInPrice = item && item.walkInPrice || 0;
    this.nePrice = item && item.nePrice || 0;

    this.category = (item && item.category) ? new Category(item.category) : undefined;
  }
}
